src/main/scala/org/ergoplatform/local/TxSender.scala

```diff
@@ -0,0 +1,101 @@
+import org.ergoplatform.nodeView.history.ErgoHistoryReader
+import org.ergoplatform.nodeView.wallet.{ErgoWallet, ErgoWalletReader}
+import org.ergoplatform.nodeView.wallet.requests._
+import org.ergoplatform.settings.{Constants, ErgoSettings}
+import scorex.core.NodeViewHolder.ReceivableMessages.LocallyGeneratedTransaction
+import scorex.core.settings.RESTApiSettings
+import scorex.util.ScorexLogging
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Actor that distributes all the coins kept in the wallet to config-defined addresses.
+  */
+class TxSender(readersHolderRef: ActorRef, nodeViewHolderRef: ActorRef, ergoSettings: ErgoSettings) extends Actor with ScorexLogging {
+
+  implicit val paymentRequestDecoder: PaymentRequestDecoder = new PaymentRequestDecoder(ergoSettings)
+  implicit val assetIssueRequestDecoder: AssetIssueRequestDecoder = new AssetIssueRequestDecoder(ergoSettings)
+  implicit val requestsHolderDecoder: RequestsHolderDecoder = new RequestsHolderDecoder(ergoSettings)
+  implicit val addressEncoder: ErgoAddressEncoder = ErgoAddressEncoder(ergoSettings.chainSettings.addressPrefix)
+  implicit val addressJsonEncoder: Encoder[ErgoAddress] = paymentRequestDecoder.addressEncoders.encoder
+  implicit val timeout: Timeout = new Timeout(60.seconds)
+
+  val settings: RESTApiSettings = ergoSettings.scorexSettings.restApi
+  val fee = PaymentRequest(Pay2SAddress(ergoSettings.chainSettings.monetary.feeProposition), ergoSettings.walletSettings.defaultTransactionFee, Seq(), Map())
+  lazy val withdrawDistribution: Map[P2PKAddress, Double] = ergoSettings
+    .walletSettings
+    .withdrawDistribution
+    .map(a => addressEncoder.fromString(a._1).get.asInstanceOf[P2PKAddress] -> a._2)
+
+  override def receive: Receive = {
+    case GetReaders =>
+      log.debug(s"Initialize TxSender with distribution ${withdrawDistribution.map(d => s"${d._1.toString()} -> ${d._2}")}")
+      readersHolderRef ! GetReaders
+
+    case Readers(h: ErgoHistoryReader, _, _, w: ErgoWalletReader) =>
+      withdraw(h, w) match {
+        case Success(_) =>
+          context.system.scheduler.scheduleOnce(10.minutes)(readersHolderRef ! GetReaders)(context.system.dispatcher)
+        case Failure(e) =>
+          log.debug(s"Failed to generate transaction: ${e.getMessage}")
+          context.system.scheduler.scheduleOnce(2.seconds)(readersHolderRef ! GetReaders)(context.system.dispatcher)
+      }
+
+    case m =>
+      log.warn(s"Unhandled message $m")
+  }
+
+  /**
+    * Will generate and broadcast transaction, that sends all coins in a wallet to addresses from withdrawDistribution.
+    *
+    * @return `Success` if transaction was sent,
+    *         `Failure` otherwise
+    */
+  private def withdraw(hr: ErgoHistoryReader, wr: ErgoWalletReader): Try[Unit] = Try {
+    val historyHeight = hr.headersHeight
+    lazy val registry = Await.result(wr.confirmedBalances, timeout.duration)
```

Comment:
Usage of `Async.await` is usually a bad practice. Why not making withdrawal in non-blocking fashion? -- wallet actor would anyway process requests synchronously.